sklearn/cross_decomposition/pls_.py

```diff
@@ -271,6 +271,10 @@ def fit(self, X, Y):
 
         # NIPALS algo: outer loop, over components
         for k in range(self.n_components):
+            if np.all(np.dot(Yk.T, Yk) < np.finfo(np.double).eps):
+                # Yk constant
+                warnings.warn('Y constant at iteration %s' % k)
+                break
```

Comment:
I would suggest the following changes:

```
diff --git a/sklearn/cross_decomposition/pls_.py b/sklearn/cross_decomposition/pls_.py
index 4ae8672..70f342d 100644
--- a/sklearn/cross_decomposition/pls_.py
+++ b/sklearn/cross_decomposition/pls_.py
@@ -271,9 +271,9 @@ class _PLS(six.with_metaclass(ABCMeta), BaseEstimator, TransformerMixin,
 
         # NIPALS algo: outer loop, over components
         for k in range(self.n_components):
-            if np.all(np.dot(Yk.T, Yk) < np.finfo(np.double).eps):
+            if np.linalg.norm(Yk) < np.finfo(np.double).eps:
                 # Yk constant
-                warnings.warn('Y constant at iteration %s' % k)
+                warnings.warn('Y residuals are null at iteration %s' % k)
                 break
             #1) weights estimation (inner loop)
             # -----------------------------------
```

@agramfort @NelleV @duchesnay does this phrasing and test seem correct to you?